### PR TITLE
list-of-versions flag regression + version id/snapshot id regression

### PR DESCRIPTION
**Pull request by siminsavani-msft**
Adding Version ID and Snapshot ID to Client Creation

### CODE REVIEW

File: ste/xfer-setProperties.go [hunk 1]
```diff
@@ -193,7 +193,18 @@ func setPropertiesFile(jptm IJobPartTransferMgr) {
 		transferDone(common.ETransferStatus.Failed(), err)
 		return
 	}
-	srcFileClient := s.NewShareClient(jptm.Info().SrcContainer).NewRootDirectoryClient().NewFileClient(jptm.Info().SrcFilePath)
+
+	sourceShare := s.NewShareClient(jptm.Info().SrcContainer)
```
Comment (gapra-msft) [inline ste/xfer-setProperties.go:197]: is it possible to set properties or delete on a share snapshot file?

Comment (siminsavani-msft) [reply ste/xfer-setProperties.go]: Share snapshot can be deleted but not modified (https://learn.microsoft.com/en-us/azure/storage/files/storage-snapshots-files#capabilities). I will update this accordingly.

Comment (gapra-msft) [reply ste/xfer-setProperties.go]: that's a share snapshot, but what about a file in the share snapshot?

File: cmd/zt_copy_blob_download_test.go [hunk 2]
```diff
@@ -955,3 +959,77 @@ func TestDryrunCopyGCPtoBlob(t *testing.T) {
...
+	// construct the raw input to simulate user input
+	rawBlobURLWithSAS := scenarioHelper{}.getRawBlobURLWithSAS(a, containerName, blobName)
+	raw := getDefaultRemoveRawInput(rawBlobURLWithSAS.String())
+	raw.recursive = true
+	raw.listOfVersionIDs = file.Name()
+	runCopyAndVerify(a, raw, func(err error) {
+		a.Nil(err)
+
+		// validate that the right number of transfers were scheduled
+		a.Equal(2, len(mockedRPC.transfers))
```
Comment (adreed-msft) [inline cmd/zt_copy_blob_download_test.go:1033]: Let's validate the data piping-- Could you check that the appropriate versions are a part of the registered transfers?

**Review by adreed-msft — COMMENTED** [review]
looks good to me.

**Review by adreed-msft — APPROVED** [review]

File: ste/xfer-setProperties.go (hunk 1 above)
Comment (siminsavani-msft) [reply ste/xfer-setProperties.go]: yeah, I guess that's not possible (https://learn.microsoft.com/en-us/rest/api/storageservices/delete-file2#remarks).

File: ste/mgr-JobPartTransferMgr.go [hunk 3]
```diff
@@ -329,7 +332,6 @@ func (jptm *jobPartTransferMgr) Info() *TransferInfo {
 	}
 
 	if versionID != "" {
-		versionID = "versionId=" + versionID
```
Comment (gapra-msft) [inline ste/mgr-JobPartTransferMgr.go:332]: line 340 and 342 below, you need to make it 

+= "&versionId=" + versionID
and then the other line
= "versionId=" + versionID


similar change with snapshot

Comment (siminsavani-msft) [reply ste/mgr-JobPartTransferMgr.go:332]: added now!

**Pull request merged**